forms-flow-documents/src/formsflow_documents/resources/pdf.py

```diff
@@ -0,0 +1,140 @@
+"""API endpoints for managing form resource."""
+import string
+from http import HTTPStatus
+
+from flask import current_app, make_response, render_template, request
+from flask_restx import Namespace, Resource
+from formsflow_api_utils.exceptions import BusinessException
+from formsflow_api_utils.utils import (
+    CLIENT_GROUP,
+    REVIEWER_GROUP,
+    auth,
+    cors_preflight,
+    profiletime,
+)
+from werkzeug.utils import secure_filename
+
+from formsflow_documents.services import PDFService
+
+API = Namespace("Form", description="Form")
+
+
+@API.route("/<string:form_id>/submission/<string:submission_id>/render", doc=False)
+class FormResourceRenderPdf(Resource):
+    """Resource to render form and submission details as html."""
+
+    @staticmethod
+    @auth.require
+    @profiletime
+    def get(form_id: string, submission_id: string):
+        """Form rendering method."""
+        pdf_service = PDFService(form_id=form_id, submission_id=submission_id)
+        default_template = "index.html"
+        template_name = request.args.get("template_name")
+        template_variable_name = request.args.get("template_variable")
+        use_template = bool(template_name)
+
+        template_name = (
+            pdf_service.url_decode(secure_filename(template_name))
+            if use_template
+            else default_template
+        )
+
+        template_variable_name = (
+            pdf_service.url_decode(secure_filename(template_variable_name))
+            if template_variable_name
+            else None
+        )
+        if not pdf_service.search_template(template_name):
+            raise BusinessException(
+                "Template not found!", HTTPStatus.INTERNAL_SERVER_ERROR
```

Comment:
Can we use BAD REQUEST (400) here than INTERNAL SERVER ERROR (500). 500 would cause API suspension by default (after a threshold) if someone uses an API manager.